osm2gtfs/creators/trips_creator.py

```diff
@@ -13,32 +19,301 @@ def __repr__(self):
+                                                         itinerary)
+
+                    # Add itinerary shape to feed.
+                    shape_id = self._add_shape_to_feed(
+                        feed, "relation/" + str(itinerary.osm_id), itinerary)
+
+                    # Add trips of each itinerary to the GTFS feed
+                    for trip_builder in prepared_trips:
+
+                        trip_builder['all_stops'] = data.get_stops()
+                        trips_count += self._add_itinerary_trips(
+                            feed, itinerary, line, trip_builder, shape_id)
+
+                # Print out status messge about added trips
+                print(" Itinerary: [" + str(itinerary.route_id) + "] " +
+                      itinerary.to.encode("utf-8") + " (added " + str(
+                          trips_count) + " trips, serving " + str(
+                              len(itinerary.get_stops())) + " stops) - " +
+                      itinerary.osm_url)
+                all_trips_count += trips_count
+
+        print("\nTotal of added trips to this GTFS: " +
+              str(all_trips_count) + "\n\n")
+        return
+
+    def _prepare_trips(self, feed, schedule, itinerary):
+        """
+        Prepare information necessary to generate trips
 
-    @staticmethod
-    def interpolate_stop_times(trip):
+        :return trips: List of different objects
         """
-        interpolate stop_times, because Navitia does not handle this itself by now
+
+        # Define a list with service days of given itinerary.
+        services = []
+        for trip in schedule['lines'][itinerary.route_id]:
+            input_fr = trip["from"]
+            input_to = trip["to"]
+            if input_fr == itinerary.fr and input_to == itinerary.to:
+                trip_services = trip["services"]
+                for service in trip_services:
+                    services.append(service)
+
+        if not services:
+            print(" Warning: From and to values didn't match with schedule.")
+
+        # Loop through all service days
+        trips = []
+        for service in services:
+
+            # Define GTFS feed service period
+            service_period = self._create_gtfs_service_period(feed, service)
+
+            # Get schedule for this itinierary's trips
+            trips_schedule = self._load_itinerary_schedule(schedule,
+                                                           itinerary, service)
+
+            # Get the stops, which are listed in the schedule
+            scheduled_stops = self._load_scheduled_stops(
+                schedule, itinerary, service)
+
+            # Prepare a trips builder container with useful data for later
+            trips.append({'service_period': service_period,
+                          'stops': scheduled_stops, 'schedule': trips_schedule})
+        return trips
+
+    def _verify_data(self, schedule, line, itinerary):
```

Comment:
Great method that you've introduced here!

Reply:
Thanks :smile: